frontend/src/components/MyApplicationItem/MyMissionItem/useMissionJudgment.ts

```diff
@@ -0,0 +1,44 @@
+import { AxiosError } from "axios";
+import { Mission, MissionStatus, Recruitment } from "../../../../types/domains/recruitments";
+import { postMyMissionJudgment } from "../../../api";
+import useTokenContext from "../../../hooks/useTokenContext";
+import { isJudgmentTimedOut } from "../../../utils/validation/judgmentTime";
+import { MISSION_STATUS } from "../../../constants/recruitment";
+import { JUDGMENT_STATUS } from "../../../constants/judgment";
+
+type MissionJudgementProps = {
+  missionItem: Mission;
+  recruitmentId: Recruitment["id"];
+};
+
+const useMissionJudgment = ({ missionItem, recruitmentId }: MissionJudgementProps) => {
+  const { token } = useTokenContext();
+
+  const { submitted, judgment, status } = missionItem;
+  const isJudgmentAvailable =
+    submitted &&
+    (judgment?.status !== JUDGMENT_STATUS.STARTED || isJudgmentTimedOut(judgment)) &&
+    !([MISSION_STATUS.ENDED, MISSION_STATUS.UNSUBMITTABLE] as MissionStatus[]).includes(status);
```

Comment:
c: `isJudgmentAvailable`의 조건문이 중요한 로직 중에 하나로 보이는데 다소 복잡해 보이는 것 같아요~! 이를 더 명확하게 분리하면 가독성도 올라가고 테스트하기도 더 좋지 않을까 싶어요!


```suggestion
   const isSubmitted = submitted;
   const isJudgmentNotStartedOrTimedOut = 
     judgment?.status !== JUDGMENT_STATUS.STARTED || isJudgmentTimedOut(judgment);
   const isMissionActive = 
     ![MISSION_STATUS.ENDED, MISSION_STATUS.UNSUBMITTABLE].includes(status);

   const isJudgmentAvailable = 
     isSubmitted && isJudgmentNotStartedOrTimedOut && isMissionActive;
```

Reply:
놓치고 있던 부분이었네요! 반영해두겠습니다💪